lib/hls/hls_parser.js

```diff
@@ -2915,7 +2915,7 @@ shaka.hls.HlsParser = class {
         if (item.name == 'EXT-X-PRELOAD-HINT') {
           partial.markAsPreload();
         }
-        if (item.getAttributeValue('INDEPENDENT') != 'YES') {
+        if (item.getAttributeValue('INDEPENDENT') != 'YES' && i > 0) {
```

Comment:
Is this supported by the spec?

Reply:
It doesn't say we can assume INDEPENDENT=YES for the first partial segment.  It does call the flag "optional", though, and I think in cases where there are no such flags on any partial segments, it is sensible to assume the first one is independent.

Can you leave a comment that it's assumed for the first, though not guaranteed by the spec?

Reply:
Sure, I'll do it tomorrow!